src/libraries/System.Private.CoreLib/src/System/BitConverter.cs

```diff
@@ -669,25 +665,24 @@ public static string ToString(byte[] value, int startIndex, int length)
             // (int.MaxValue / 3) == 715,827,882 Bytes == 699 MB
             ArgumentOutOfRangeException.ThrowIfGreaterThan(length, int.MaxValue / 3);
 
-            return string.Create(length * 3 - 1, (value, startIndex, length), static (dst, state) =>
+            int stringLength = length * 3 - 1;
+            string result = string.FastAllocateString(stringLength);
+
+            var dst = new Span<char>(ref result.GetRawStringData(), stringLength);
```

Comment:
Just to make it a bit more obvious in the span ctor that the length is correct:
```suggestion
            string result = string.FastAllocateString(length * 3 - 1);

            var dst = new Span<char>(ref result.GetRawStringData(), result.Length);
```